src/node_api.cc

```diff
@@ -132,7 +132,9 @@ class ThreadSafeFunction : public node::AsyncResource {
       finalize_cb(finalize_cb_),
       call_js_cb(call_js_cb_ == nullptr ? CallJs : call_js_cb_),
       handles_closing(false) {
-    ref.Reset(env->isolate, func);
+    if (!func.IsEmpty()) {
```

Comment:
You don't need this check, `ref.Reset(isolate, func)` where `func.IsEmpty() == true` is fine, it's a no-op.